### PR TITLE
Newsletters Module

### PR DESCRIPTION
**Pull request by stryker2k2**
### Description:
- Created a new module called 'newsletters' that will scrap the websites (or recursive websites, thanks to BBOT's sub-domain enumeration) looking for the presence of an 'email type' that also contains a 'placeholder'. The combination of these two HTML items usually signify the presence of an "Enter Your Email Here" type Newsletter Subscription service. This module could be used to find newsletters for a future email bombing attack and/or find user-input fields that could be be susceptible to overflows or injections.

### Ticket/Issue
- https://github.com/blacklanternsecurity/bbot/issues/1056

### To Review:
- [x] Run BBOT Module on a website without a Newsletter (ex: blacklanternsecurity.com)
    - `bbot -m httpx newsletters -t blacklanternsecurity.com`
    - Expect 0 Newsletter Events
- [x] Run BBOT Module on a website WITH a Newsletter (ex: futureparty.com)
    - `bbot -m httpx newsletters -t futureparty.com`
    - Expect 1 or more Newsletter Events
- [x] View output on neo4j to ensure it displays well
- [x] Code Review
- [x] Tests Review

### Notes:
- Must be ran with with 'httpx'
- `bbot -m httpx newsletters -t <insert target here>`

### CODE REVIEW

**Comment by stryker2k2** [comment]
I will be resolving the merge conflicts (via command line) as well as modifying the test_module_newsletters. I'd like to make two checks, one for WITH newsletters and one for WITHOUT newsletters. I currently have one check that fails if either one doesn't pass.

File: bbot/modules/newsletters.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# https://www.milkkarten.net/
+# https://geekout.mattnavarra.com/
+
+deps_pip = ["beautifulsoup4"]
+
+
+class newsletters(BaseModule):
+    watched_events = ["HTTP_RESPONSE"]
+    produced_events = ["FINDING"]
+    flags = ["passive", "safe"]
```
Comment (TheTechromancer) [inline bbot/modules/newsletters.py]: "passive" here should be "active" since it relies on web requests.

File: lint-tests.sh
Comment (TheTechromancer) [inline lint-tests.sh:1]: We already have a file for this, `bbot/test/run_tests.sh`.

**Review by TheTechromancer — CHANGES_REQUESTED** [review]

**Review by TheTechromancer — APPROVED** [review]

**Comment by stryker2k2** [comment]
Yippie!

**Pull request merged**